model/task_queue.go

```diff
@@ -372,24 +382,54 @@ func updateTaskQueue(distro string, taskQueue []TaskQueueItem, distroQueueInfo D
 	return errors.WithStack(err)
 }
 
-func ClearTaskQueue(distro string) error {
+func ClearTaskQueue(distroId string) error {
 	grip.Info(message.Fields{
 		"message": "clearing task queue",
-		"distro":  distro,
+		"distro":  distroId,
 	})
+
+	catcher := grip.NewBasicCatcher()
+
+	distroQueueInfo, err := GetDistroQueueInfo(distroId)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	distroQueueInfo.Length = 0
+	distroQueueInfo.CountOverThreshold = 0
```

Comment:
This should also clear expected duration and task group infos shouldn't it?

Reply:
I wanted to err on the conservative side/ not blanking things but if you think that's advisable I have not specific objection,

Reply:
I believe it should be fine to clear the task group info and will prevent some odd data inconsistencies in the task queue. I'm pretty sure it's used primarily for the host allocator to correctly figure out how many hosts have to be created (but that shouldn't matter if we're already clearing the task queue).